### PR TITLE
Add `ERC7984IdentityCheck` extension

### PR DESCRIPTION
**Pull request by arr00**
## Summary by CodeRabbit

* **New Features**
  * Added ERC7984 extension that enforces recipient identity verification during token transfers, reverting if the recipient is not verified in the configured identity registry.

* **Tests**
  * Added comprehensive test suite covering constructor validation, identity registry integration, transfer behavior, and edge cases.

* **Chores**
  * Updated Changeset for upcoming minor version release.

[Review Change Stack](https://app.coderabbit.ai/change-stack/OpenZeppelin/openzeppelin-confidential-contracts/pull/364)

### CODE REVIEW

File: contracts/token/ERC7984/extensions/ERC7984IdentityCheck.sol [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    /// @dev See {ERC7984-_update}. Performs identity check before updating the balance.
+    function _update(address from, address to, euint64 amount) internal virtual override returns (euint64) {
+        if (to != address(0) && !IIdentityRegistry(_identityRegistry).isVerified(to)) {
+            revert ERC7984InvalidIdentity(to);
+        }
+        return super._update(from, to, amount);
+    }
+
+    /// @dev Returns the address of the identity registry.
+    function identityRegistry() public view returns (address) {
```
Comment (arr00) [inline contracts/token/ERC7984/extensions/ERC7984IdentityCheck.sol]:
```suggestion
    function identityRegistry() public virtual view returns (address) {
```

File: contracts/token/ERC7984/extensions/ERC7984IdentityCheck.sol [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+    constructor(address identityRegistry_) {
+        _setIdentityRegistry(identityRegistry_);
+    }
+
+    /// @dev See {ERC7984-_update}. Performs identity check on the recipient before updating the balance.
+    function _update(address from, address to, euint64 amount) internal virtual override returns (euint64) {
+        if (to != address(0) && !IIdentityRegistry(_identityRegistry).isVerified(to)) {
+            revert ERC7984InvalidIdentity(to);
+        }
```
Comment (Amxx) [inline contracts/token/ERC7984/extensions/ERC7984IdentityCheck.sol:35]: this could ahve been

```suggestion
        require(to == address(0) || IIdentityRegistry(_identityRegistry).isVerified(to), ERC7984InvalidIdentity(to));
```

File: contracts/token/ERC7984/extensions/ERC7984IdentityCheck.sol [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    /// @dev Returns the address of the identity registry.
+    function identityRegistry() public view virtual returns (address) {
+        return _identityRegistry;
+    }
+
+    /// @dev Sets the identity registry. Inheriting contracts are responsible for access control.
+    function _setIdentityRegistry(address identityRegistry_) internal virtual {
+        require(
+            identityRegistry_ != address(0) && identityRegistry_.code.length != 0,
```
Comment (Amxx) [inline contracts/token/ERC7984/extensions/ERC7984IdentityCheck.sol:47]: This would be enough as it will catpure address(0) having no code.
```suggestion
            identityRegistry_.code.length != 0,
```
